File: unidbg-api/src/main/java/com/vortexdbg/unix/FileListener.java
package com.vortexdbg.unix;

import com.vortexdbg.Emulator;
import com.vortexdbg.file.FileIO;

public interface FileListener {

    void onOpenSuccess(Emulator<?> emulator, String pathname, FileIO io);

    void onRead(Emulator<?> emulator, String pathname, byte[] bytes);
    void onWrite(Emulator<?> emulator, String pathname, byte[] bytes);

    void onClose(Emulator<?> emulator, FileIO io);

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/unix/FileListener.kt
package com.vortexdbg.unix

import com.vortexdbg.Emulator
import com.vortexdbg.file.FileIO

interface FileListener {

    fun onOpenSuccess(emulator: Emulator<*>, pathname: String, io: FileIO)

    fun onRead(emulator: Emulator<*>, pathname: String, bytes: ByteArray)
    fun onWrite(emulator: Emulator<*>, pathname: String, bytes: ByteArray)

    fun onClose(emulator: Emulator<*>, io: FileIO)

}
